openff/interchange/smirnoff/_create.py

```diff
@@ -93,17 +97,61 @@ def validate_topology(value):
         )
 
 
+def _preprocess_preset_charges(
+    molecules_with_preset_charges: list[Molecule] | None,
+) -> list[Molecule] | None:
+    """
+    Pre-process the molecules_with_preset_charges argument.
+
+    If molecules_with_preset_charges is None, return None.
+
+    If molecules_with_preset_charges is list[Molecule], ensure that
+
+    1. The input is a list of Molecules
+    2. Each molecule has assign partial charges
+    3. Ensure no molecules are isomorphic with another in the list
+
+    """
+    if molecules_with_preset_charges is None:
+        return None
+
+    molecule_set = {molecule.to_smiles() for molecule in molecules_with_preset_charges}
+
+    if len(molecule_set) != len(molecules_with_preset_charges):
+        raise PresetChargesError(
+            "All molecules in the molecules_with_preset_charges list must be isomorphically unique from each other",
+        )
+
+    for molecule in molecules_with_preset_charges:
+        if molecule.partial_charges is None:
+            raise PresetChargesError(
+                "All molecules in the molecules_with_preset_charges list must have partial charges assigned.",
+            )
+
+    return molecules_with_preset_charges
+
+
 def _create_interchange(
     force_field: ForceField,
     topology: Topology | list[Molecule],
     box: Quantity | None = None,
     positions: Quantity | None = None,
-    charge_from_molecules: list[Molecule] | None = None,
+    molecules_with_preset_charges: list[Molecule] | None = None,
     partial_bond_orders_from_molecules: list[Molecule] | None = None,
     allow_nonintegral_charges: bool = False,
 ) -> Interchange:
+    molecules_with_preset_charges = _preprocess_preset_charges(molecules_with_preset_charges)
+
     _check_supported_handlers(force_field)
 
+    if molecules_with_preset_charges is not None and "VirtualSites" in force_field.registered_parameter_handlers:
+        warnings.warn(
+            "Preset charges were provided (via `charge_from_molecules`) alongside a force field that includes "
+            "virtual site parameters. Note that virtual sites will be applied charges from the force field and "
+            "cannot be given preset charges.",
```

Comment:
```suggestion
            "Preset charges were provided (via `charge_from_molecules`) alongside a force field that includes "
            "virtual site parameters. Note that virtual sites will be applied charges from the force field and "
            "cannot be given preset charges. Virtual sites may also affect the charges of their orientation "
            "atoms, even if those atoms are given preset charges.",
```

Do you think adding something like this to the warning will be useful for people? It seems like the sort of thing that would surprise me but maybe it's just noise.

Reply:
I think this is a great example of something that seems repetitive / doesn't need to be made explicit to me (since I'm so steeped in the details) but is worth repeating / making explicit to a user